### DISCUSSION TITLE
Integrate with Intervals.icu API

### DISCUSSION THREAD

**Discussion by Lunchtime0614** [doudar/SmartSpin2k#509] [Ideas]
I'm wondering if any of the maintainers have thought of integrating with the https://intervals.icu api (https://intervals.icu/api/v1/docs/swagger-ui/index.html) whereby SS2k could pull the workout/event for the current day or maybe even show a minimalist calendar for the current week...then have SS2k run the training program (zwo file) itself.  Basically using the zwo file as it's source of commands rather than a different training app (which I do today through trainer day, etc).

**Comment by doudar** [comment]
It wouldn't be difficult to write a simple JavaScript webpage to do this since you can set watt targets via http get requests. I've not documented the API, but if you look at (the server.on's) in http_server.cpp and/or streamfit.html, you'll see how to set the target points.

**Reply by Lunchtime0614** [reply]
Thanks for the reply, unfortunately software engineering is not my expertise, so while I can lookup the files you reference, I do not have the skills to implement such a function.
